chirp/drivers/baofeng_digital.py

```diff
@@ -0,0 +1,492 @@
+                    f'Radio replied incorrectly to preamble step {step}')
+
+    def _clean_buffer(self):
+        self.pipe.timeout = 0.001
+        self.pipe.read(256)
+        self.pipe.timeout = 0.1
+
+    def _exit_programming(self):
+        try:
+            self.pipe.write(CMD_EXIT)
+        except Exception:
+            pass
+
+    def sync_in(self):
+        try:
+            self._enter_programming(False)
+
+            memory = memmap.MemoryMapBytes(b"\xFF" * self._memsize)
+            for chunk in self._proto.mem_chunks:
+                for addr in range(chunk.start, chunk.end, 8):
+                    self.pipe.write(struct.pack('>cHB', CMD_READ, addr, 8))
+                    reply = self.pipe.read(12)
+                    if len(reply) != 12:
+                        raise errors.RadioError(
+                            f'Failed to read block at 0x{addr:03x}')
+
+                    self.pipe.write(CMD_ACK)
+                    if self.pipe.read(1) != CMD_ACK:
+                        raise errors.RadioError(
+                            f'No ACK after reading from 0x{addr:03x}')
+
+                    memory.set(addr, reply[4:])
+        except errors.RadioError:
+            raise
+        except Exception as e:
+            raise errors.RadioError(
+                f'Failed to communicate with the radio: {e}')
+        finally:
+            self._exit_programming()
+
+        self._mmap = memory
+        self.process_mmap()
+
+    def sync_out(self):
+        try:
+            self._enter_programming(True)
+
+            for chunk in self._proto.mem_chunks:
+                if chunk.read_only:
+                    continue
+
+                for addr in range(chunk.start, chunk.end, 8):
+                    frame = (
+                        struct.pack('>cHB', CMD_WRITE, addr, 8) +
+                        self._mmap.get(addr, 8)
+                    )
+                    self.pipe.write(frame)
+
+                    if self.pipe.read(1) != CMD_ACK:
+                        raise errors.RadioError(
+                            f'Not ACK after writing to 0x{addr:03x}')
+        except errors.RadioError:
+            raise
+        except Exception as e:
+            raise errors.RadioError(
+                f'Failed to communicate with the radio: {e}')
+        finally:
+            self._exit_programming()
+
+    def process_mmap(self):
+        mem_fmt = self.MEM_FORMAT.format(chan_count=self._proto.chan_count)
+        self._memobj = bitwise.parse(mem_fmt, self._mmap)
+
+    def get_raw_memory(self, number):
+        return repr(self._memobj.memory[number])
+
+    def _parse_tone(self, value):
+        if value in (0x0000, 0xFFFF):
+            return ('', None, 'N')
+
+        # Parse lower 14 bit as BCD
+        bcd_value = (
+            1000 * (value >> 12 & 0x3) +
+            100 * (value >> 8 & 0xF) +
+            10 * (value >> 4 & 0xF) +
+            (value & 0xF)
+        )
```

Comment:
Can you re-use `bitwise.bcd_to_int()`?

Reply:
...and I assume you can't use a bcd bitwise type for some reason? I figured you were setting it as a non-BCD value somewhere  but I don't see it.